sources/Controller/Newsroom/iTopNewsroomController.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+use Combodo\iTop\Application\Branding;
+use Combodo\iTop\Application\TwigBase\Controller\Controller;
+
+class iTopNewsroomController extends Controller
+{
+	public const ROUTE_NAMESPACE = 'itopnewsroom';
+
+	/**
+	 * @return \iTopWebPage
+	 * @throws \ApplicationException
+	 * @throws \CoreException
+	 * @throws \OQLException
+	 */
+	public function OperationViewAll()
+	{
+		$oPage = new iTopWebPage('cc');
+		$oSearch = DBObjectSearch::FromOQL('SELECT EventiTopNotification WHERE read = "no"');
+		$oSearch->AddCondition('contact_id', UserRights::GetContactId(), '=');
+		$oBlock = new DisplayBlock($oSearch, 'search', false /* Asynchronous */, []);
+		$oBlock->Display($oPage, 0);
+		$oPage->add("<div class='sf_results_area ibo-add-margin-top-250' data-target='search_results'>");
+		return $oPage;
+	}
+
+	/**
+	 * @return \AjaxPage
+	 * @throws \ArchivedObjectException
+	 * @throws \CoreException
+	 * @throws \CoreUnexpectedValue
+	 * @throws \MySQLException
+	 * @throws \OQLException
+	 */
+	public function OperationFetchUnreadMessages()
+	{
+		$sCallback = utils::ReadParam('callback', '');
+		$oPage = new AjaxPage('');
+
+		$aMessages = [];
+		$iContactId = UserRights::GetContactId();
+		
+		if($iContactId !== '') {
```

Comment:
```suggestion
		if (\utils::IsNotNullOrEmptyString($iContactId)) {
```

`null` can be returned as well